### PR TITLE
feat(core): implement oa disputer whitelist in ssm

### PR DESCRIPTION
**Pull request by Reinis-FRP**
**Motivation**

Some users of Optimistic Assertor might want to allow only whitelisted disputer set.

**Summary**

Adds Sovereign Security Manager checking if disputer is whitelisted.

**Details**

Adds new `isDisputeAllowed` method for all SSM (defaults to true) that is called by Optimistic Assertor during dispute.

**Testing**

Check a box to describe how you tested these changes and list the steps for reviewers to test.

- [ ]  Ran end-to-end test, running the code as in production
- [x]  New unit tests created
- [ ]  Existing tests adequate, no new tests required
- [ ]  All existing tests pass
- [ ]  Untested

### CODE REVIEW

File: packages/core/contracts/optimistic-assertor/implementation/OptimisticAssertor.sol [hunk 1]
```diff
@@ -268,6 +269,12 @@ contract OptimisticAssertor is Lockable, OptimisticAssertorInterface, Ownable {
         return SovereignSecurityManagerInterface(ssm).getAssertionPolicies(assertionId);
     }
 
+    function _isDisputeAllowed(bytes32 assertionId) internal view returns (bool) {
+        address ssm = assertions[assertionId].sovereignSecurityManager;
+        if (ssm == address(0)) return true;
+        return SovereignSecurityManagerInterface(ssm).isDisputeAllowed(assertionId, msg.sender);
```
Comment (Reinis-FRP) [inline packages/core/contracts/optimistic-assertor/implementation/OptimisticAssertor.sol:275]: Note that this adds gas costs for all SSM implementations even if not using disputer whitelist. Alternative would be to store additional SSM policy on whether to call `isDisputeAllowed` at assertion time.

Comment (chrismaree) [reply packages/core/contracts/optimistic-assertor/implementation/OptimisticAssertor.sol:275]: we could also add add a prop at assertion time to the policy whether the OA should check with the asserter on `isDisputeAllowed`. so it's off by default unless you explicitly want to check this.

**Review by chrismaree — COMMENTED** [review]
This PR has gas implications and generally adds a bit more bloat to the OA design. I think we should leave this one until a bit later in the design process(get the others in first) and then circle back to this. 

I think it could be a good option BUT we should not bloat the dispute callback in all cases. it might be a good idea to add this to the policy set by the SSM at assertion time and use this to check if the SSM whats to enable/disable disputes, in the same way the SSM decides on how to arbitrate & unplug.

**Comment by chrismaree** [comment]
I rebased this against master. the code was out of date vs the base OA.

File: packages/core/contracts/optimistic-assertor/implementation/OptimisticAssertor.sol (hunk 1 above)
Comment (chrismaree) [reply packages/core/contracts/optimistic-assertor/implementation/OptimisticAssertor.sol:318]: wdyt? this could be in `ssmSettings`?

Comment (Reinis-FRP) [reply packages/core/contracts/optimistic-assertor/implementation/OptimisticAssertor.sol:322]: Added `validateDisputers` to SSM policies that is checked before calling `isDisputeAllowed`. By default this `validateDisputers` is disabled and used only in this SSM implementation

**Review by chrismaree — APPROVED** [review]
I like this! good job @Reinis-FRP

**Pull request merged**